src/main/java/net/darkhax/bookshelf/common/BookshelfRegistry.java

```diff
@@ -1,198 +1,260 @@
+     *                     least one for this to work.
+     * @param output       The ItemStack to be created by this recipe.
      */
-    public static void addAnvilRecipe (ItemStack inputLeft, ItemStack inputRight, String requiredName, int experience, ItemStack output) {
-        
+    public static void addAnvilRecipe(ItemStack inputLeft, ItemStack inputRight, String requiredName, int experience, ItemStack output) {
+
         anvilRecipes.add(new AnvilRecipe(inputLeft, inputRight, requiredName, experience, 0, output));
     }
-    
+
     /**
      * Adds a new AnvilRecipe to the registry. Inputs can be null.
-     * 
-     * @param inputLeft The ItemStack required in the left slot of the Anvil GUI.
-     * @param inputRight The ItemStack required in the left slot of the Anvil GUI.
+     *
+     * @param inputLeft    The ItemStack required in the left slot of the Anvil GUI.
+     * @param inputRight   The ItemStack required in the left slot of the Anvil GUI.
      * @param requiredName A name requirement for this recipe to complete.
-     * @param experience The amount of experience levels this recipe will cost. Must be at
-     *        least one for this to work.
+     * @param experience   The amount of experience levels this recipe will cost. Must be at
+     *                     least one for this to work.
      * @param materialCost The amount of items to consume from the right side of the Anvil GUI.
-     *        If 0, all will be used.
-     * @param output The ItemStack to be created by this recipe.
+     *                     If 0, all will be used.
+     * @param output       The ItemStack to be created by this recipe.
      */
-    public static void addAnvilRecipe (ItemStack inputLeft, ItemStack inputRight, String requiredName, int experience, int materialCost, ItemStack output) {
-        
+    public static void addAnvilRecipe(ItemStack inputLeft, ItemStack inputRight, String requiredName, int experience, int materialCost, ItemStack output) {
+
         anvilRecipes.add(new AnvilRecipe(inputLeft, inputRight, requiredName, experience, materialCost, output));
     }
-    
+
     /**
      * Adds a new AnvilRecipe to the registry directly.
-     * 
+     *
      * @param recipe The AnvilRecipe to register.
      */
-    public void addAnvilRecipe (AnvilRecipe recipe) {
-        
+    public void addAnvilRecipe(AnvilRecipe recipe) {
+
         anvilRecipes.add(recipe);
     }
-    
+
+    @SideOnly(Side.CLIENT)
+    public static void addRenderItem(Item item, IItemRenderer itemRenderer) {
+
+        List<IItemRenderer> rendererIntegerMap = itemRenderList.get(item);
+        if (rendererIntegerMap == null) {
+
+            rendererIntegerMap = new ArrayList<>();
+            itemRenderList.put(item, rendererIntegerMap);
+        }
+        rendererIntegerMap.add(itemRenderer);
+    }
+
+    @SideOnly(Side.CLIENT)
+    public static void renderPreModel(ItemStack stack, IBakedModel model) {

```

Comment:
I would prefer these methods/hooks not be part of the registry class. They should probably go in the transformer, or a separate BookshelfHooks class.